<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QtExample</class>
 <widget class="QMainWindow" name="QtExample">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>293</width>
    <height>601</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QtExample</string>
  </property>
  <widget class="QWidget" name="central_widget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QGroupBox" name="group_data_generation">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Data generation</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Model types:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkbox_line">
         <property name="text">
          <string>Line</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkbox_sphere">
         <property name="text">
          <string>Sphere</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Points per model:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinbox_ppm">
         <property name="minimum">
          <number>100</number>
         </property>
         <property name="maximum">
          <number>10000</number>
         </property>
         <property name="value">
          <number>1000</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Inlier fraction:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="spinbox_inlier_fraction">
         <property name="minimum">
          <double>0.010000000000000</double>
         </property>
         <property name="maximum">
          <double>1.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
         <property name="value">
          <double>0.700000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QGroupBox" name="group_model_fitting">
      <property name="title">
       <string>Model fitting</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QLabel" name="label_model_type">
         <property name="text">
          <string>Model type:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radio_line">
         <property name="text">
          <string>Line</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radio_plane">
         <property name="text">
          <string>Plane</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="radio_sphere">
         <property name="text">
          <string>Sphere</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_distance_threshold">
         <property name="toolTip">
          <string>Distance to model threshold</string>
         </property>
         <property name="text">
          <string>Distance threshold:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="spinbox_distance_threshold">
         <property name="minimum">
          <double>0.010000000000000</double>
         </property>
         <property name="maximum">
          <double>10.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_probability">
         <property name="toolTip">
          <string>Desired probability of choosing at least one sample free from outliers</string>
         </property>
         <property name="text">
          <string>Probability:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="spinbox_probability">
         <property name="minimum">
          <double>0.010000000000000</double>
         </property>
         <property name="maximum">
          <double>0.990000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.010000000000000</double>
         </property>
         <property name="value">
          <double>0.990000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_max_iterations">
         <property name="toolTip">
          <string>Maximum number of iterations</string>
         </property>
         <property name="text">
          <string>Max iterations:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinbox_max_iterations">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>10000</number>
         </property>
         <property name="singleStep">
          <number>10</number>
         </property>
         <property name="value">
          <number>1000</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="button_fit">
         <property name="text">
          <string>Fit</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="0">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="0" column="1" rowspan="3">
     <widget class="TViewerWidget" name="viewer" native="true"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menu_bar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>293</width>
     <height>19</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionAction_group">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Action group</string>
   </property>
   <property name="shortcut">
    <string>A</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>TViewerWidget</class>
   <extends>QWidget</extends>
   <header>tviewer/tviewer_widget/tviewer_widget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>spinbox_inlier_fraction</sender>
   <signal>valueChanged(double)</signal>
   <receiver>QtExample</receiver>
   <slot>onDataGenerationSettingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>76</x>
     <y>229</y>
    </hint>
    <hint type="destinationlabel">
     <x>146</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinbox_ppm</sender>
   <signal>valueChanged(int)</signal>
   <receiver>QtExample</receiver>
   <slot>onDataGenerationSettingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>76</x>
     <y>177</y>
    </hint>
    <hint type="destinationlabel">
     <x>146</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkbox_line</sender>
   <signal>stateChanged(int)</signal>
   <receiver>QtExample</receiver>
   <slot>onDataGenerationSettingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>76</x>
     <y>103</y>
    </hint>
    <hint type="destinationlabel">
     <x>146</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkbox_sphere</sender>
   <signal>stateChanged(int)</signal>
   <receiver>QtExample</receiver>
   <slot>onDataGenerationSettingsChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>76</x>
     <y>129</y>
    </hint>
    <hint type="destinationlabel">
     <x>146</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_fit</sender>
   <signal>clicked()</signal>
   <receiver>QtExample</receiver>
   <slot>onButtonFitClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>76</x>
     <y>365</y>
    </hint>
    <hint type="destinationlabel">
     <x>146</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onDataGenerationSettingsChanged()</slot>
  <slot>onButtonFitClicked()</slot>
 </slots>
</ui>
